handlers/validator.go

```diff
@@ -2006,152 +2006,157 @@ func ValidatorSync(w http.ResponseWriter, r *http.Request) {
 
 	// retrieve all sync periods for this validator
 	var syncPeriods []uint64 = []uint64{}
-
 	err = db.ReaderDb.Select(&syncPeriods, `
 		SELECT period
 		FROM sync_committees 
 		WHERE validatorindex = $1
 		ORDER BY period desc`, validatorIndex)
+
 	if err != nil {
 		utils.LogError(err, "error getting sync tab count data of sync-assignments from db", 0, errFields)
 		http.Error(w, "Internal server error", http.StatusInternalServerError)
 		return
 	}
 
-	// total count of sync duties for this validator
-	totalCount := uint64(0)
-
 	tableData := [][]interface{}{}
 
-	if len(syncPeriods) > 0 {
+	data := &types.DataTableResponse{
+		Draw:            draw,
+		RecordsTotal:    0,
+		RecordsFiltered: 0,
+		Data:            tableData,
+	}
 
-		latestProposedSlot := services.LatestProposedSlot()
+	if len(syncPeriods) == 0 {
+		// no sync periods for this validator => early exit with empty tableData
```

Comment:
**issue:** There is no early return here in case of empty `syncPeriods` as was inteded.
You just return in case there is an encode error.

If you navigate to e.g. `/validator/428593/sync?draw=1&start=0&length=10` for a validator without syncs on Prater you get past this point and trigger a panic at https://github.com/gobitfly/eth2-beaconchain-explorer/blob/dd8e64b12f9902c7fc750cfc99150c660b8002f2/handlers/validator.go#L2079 because `slots` is empty.

The same line also triggers a panic if the user searches for entries that lie beyond the end.
E.g. `/validator/31484/sync?draw=10001&start=100000&length=10` (on Prater) has 2 sync and therefore `len(slots) = 16384`. As a result `slots[100000]` crashes.

Reply:
Jeah, that's because I did not implement an "Early Return" but only an "Early" i.e. an "Early Return" without the proper "Return".
I am sorry, I will fix that quite soon. Shame on me.

Reply:
Fixed.